docs: virtual-list docs and examples (#538)
vaadin/docs: articles/ds/components/virtual-list/index.asciidoc

---
title: Virtual List
layout: tabbed-page
tab-title: Usage
page-links:
  - https://github.com/vaadin/flow-components/releases/tag/{moduleMavenVersion:com.vaadin:vaadin-virtual-list-flow}[Flow {moduleMavenVersion:com.vaadin:vaadin-virtual-list-flow}]
  - https://github.com/vaadin/web-components/releases/tag/v{moduleNpmVersion:vaadin-virtual-list}[Web Component {moduleNpmVersion:vaadin-virtual-list}]
---
= Virtual List

> :components-path-prefix: ../

Reviewer: Is this attribute used anywhere on this page?

// tag::description[]
Virtual list displays a vertically aligned set of items inside a scrollable container.

> Each item is rendered per need basis as the user scrolls the list, which makes the component especially suitable for displaying long item lists without sacrificing performance.

Reviewer 2: ```suggestion
Each item is rendered on the fly as the user scrolls the list, which makes the component especially suitable for displaying long lists of items without sacrificing performance.
```
I wrote "rendered" here, but should it rather be "generated"? Is the actual data generated or just rendering?

Author: Rendered is better since that's what it's about

// end::description[]

> To use the component, you need to assign it a set of data items and a renderer that's used for rendering each individual data item.

Reviewer 2: ```suggestion
To use the component, you need to assign it a set of data items and a _renderer_ that is used for rendering each individual data item.
```

The height of an item is determined by its content and can change dynamically.

> [.example]

Reviewer 2: The example doesn't include the sources for the data source, and it is not trivial to figure out what they should be like.

Author: Aligned with other similar examples and included the Person.java / Person.ts files
